atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultSweepAssignedBucketStoreTest.java

```diff
@@ -92,4 +101,65 @@ public void updateStateMachineForBucketAssignerModifiesStateToNewIfOriginalMatch
         store.updateStateMachineForBucketAssigner(initialState, newState);
         assertThat(store.getBucketStateAndIdentifier()).isEqualTo(newState);
     }
+
+    @Test
+    public void getStartingBucketsForShardReturnsStartingBucketWhenSet() {
+        List<Bucket> buckets = List.of(
+                Bucket.of(ShardAndStrategy.of(12, SweeperStrategy.THOROUGH), 512),
+                Bucket.of(ShardAndStrategy.of(54, SweeperStrategy.CONSERVATIVE), 154389),
+                Bucket.of(ShardAndStrategy.of(25, SweeperStrategy.NON_SWEEPABLE), 97312907));
+
+        buckets.forEach(store::updateStartingBucketForShardAndStrategy);
+
+        assertThat(store.getStartingBucketsForShards(
+                        buckets.stream().map(Bucket::shardAndStrategy).collect(Collectors.toSet())))
+                .containsExactlyInAnyOrderElementsOf(buckets);
+    }
+
+    @Test
+    public void getStartingBucketsForShardReturnsZeroBucketIdentifierForUnsetShards() {
+        Bucket bucket = Bucket.of(ShardAndStrategy.of(12, SweeperStrategy.THOROUGH), 512);
+        Bucket unsetBucket = Bucket.of(ShardAndStrategy.of(54, SweeperStrategy.CONSERVATIVE), 0);
+
+        store.updateStartingBucketForShardAndStrategy(bucket);
+
+        assertThat(store.getStartingBucketsForShards(Set.of(bucket.shardAndStrategy(), unsetBucket.shardAndStrategy())))
+                .containsExactlyInAnyOrder(bucket, unsetBucket);
+    }
+
+    @Test
+    public void updateStartingBucketForShardDoesNotDecreaseExistingValue() {
+        Bucket existingBucket = Bucket.of(ShardAndStrategy.of(12, SweeperStrategy.THOROUGH), 512);
+        Bucket newBucket = Bucket.of(ShardAndStrategy.of(12, SweeperStrategy.THOROUGH), 256);
+
+        store.updateStartingBucketForShardAndStrategy(existingBucket);
+        store.updateStartingBucketForShardAndStrategy(newBucket);
+
+        assertThat(store.getStartingBucketsForShards(Set.of(existingBucket.shardAndStrategy())))
+                .containsExactly(existingBucket);
+    }
+
+    @Test
+    public void updateStartingBucketForShardSetsToAtLeastExistingValue() {
```

Comment:
I think this one is more about setting to at least the new value? The shard-and-strategy for both existing and new are the same here